src/background/extension.ts

```diff
@@ -237,7 +237,7 @@ export class Extension {
             collect: async () => {
                 return await this.collectData();
             },
-            changeSettings: (settings) => this.changeSettings(settings),
+            changeSettings: async (settings) => this.changeSettings(settings),
```

Comment:
Now this function is async and its promise resolves once all settings are saved.